### ISSUE TITLE
Does the arbiter tree and mask logic overlap in rr_arb_tree ?

### ISSUE THREAD

**Issue by Hu1chen** [pulp-platform/common_cells#245]
I think the lower_idx or upper_idx(#209, 210) is the next port to be executed, why add arbitration tree(#254-)? 
is this redundant or there is a problem in my understanding?

**Comment by phsauter** [comment]
No, they are not redundant: `upper_idx`/`lower_idx` only compute the next round-robin priority state after a successful grant.  
The arbitration tree still computes the current winner, selected data/index, and grant routing from current `rr_q`.
